nvme.c

```diff
@@ -7364,13 +7364,17 @@ static int dsm(int argc, char **argv, struct command *acmd, struct plugin *plugi
 			return err;
 		}
 	}
-	if (!cfg.cdw11)
-		cfg.cdw11 = (cfg.ad << 2) | (cfg.idw << 1) | (cfg.idr << 0);
+	if (cfg.cdw11) {
+		cfg.ad = NVME_GET(cfg.cdw11, DSM_CDW11_AD);
+		cfg.idw = NVME_GET(cfg.cdw11, DSM_CDW11_IDW);
+		cfg.idr = NVME_GET(cfg.cdw11, DSM_CDW11_IDR);
+	}
 
 	dsm = nvme_alloc(sizeof(*dsm) * 256);
 	if (!dsm)
 		return -ENOMEM;
 
+	nvme_init_dsm_range(dsm, ctx_attrs, nlbs, slbas, nr);
```

Comment:
~~`nlbs` is not correctly wired up with the `struct config` above.
The `char *blocks;` member of `struct config` must be `char *nlbs;` .~~
Edit: Sorry, it is correctly wired up with the `argconfig_parse_comma_sep_array_u32` helper.

I think we should rename the `struct config` members according to the spec to avoid further confusion :)